src/it/java/net/spy/memcached/ClientBaseCase.java

```diff
@@ -140,7 +143,13 @@ protected void restoreClusterConfig(String config, Collection<NodeEndPoint> endp
 	if(!endpoints.isEmpty() && config != null) {
 	  ArrayList<MemcachedClient> clients = new ArrayList<MemcachedClient>();
       for (NodeEndPoint endpoint : endpoints) {
-        MemcachedClient currentClient = new MemcachedClient(endpoint.getInetSocketAddress());
+        MemcachedClient currentClient;
+        if (!TestConfig.isTlsMode()){
+          currentClient = new MemcachedClient(endpoint.getInetSocketAddress());
+        } else {
+          List<InetSocketAddress> addrs = AddrUtil.getAddresses(endpoint.getIpAddress());
+          currentClient = staticMemcachedClient(addrs);
+        }
```

Comment:
Not sure why this code logic needs to be changed. Auto-discovery/dynamic mode client should not depend on TLS or not.

Reply:
For non-tls, we can create a normal client (without specifying Client mode) and set configuration to the local server. But for TLS mode, we need to create a Static client first, set configuration to the local server for pretending ElastiCache cluster server. Then the rest will be the same with non-tls mode. Line 153 explains why I did it.